Classes/Controller/Embedded3dViewerController.php

```diff
@@ -0,0 +1,97 @@
+<?php
+/**
+ * (c) Kitodo. Key to digital objects e.V. <contact@kitodo.org>
+ *
+ * This file is part of the Kitodo and TYPO3 projects.
+ *
+ * @license GNU General Public License version 3 or later.
+ * For the full copyright and license information, please read the
+ * LICENSE.txt file that was distributed with this source code.
+ */
+
+namespace Kitodo\Dlf\Controller;
+
+use Kitodo\Dlf\Common\AbstractDocument;
+
+/**
+ * Plugin 'Embedded3dViewer' for the 'dlf' extension
+ *
+ * @package TYPO3
+ * @subpackage dlf
+ *
+ * @access public
+ */
+class Embedded3dViewerController extends AbstractController
+{
+
+    const MIDDLEWARE_DLF_EMBEDDED_3D_VIEWER_PREFIX = '/?middleware=dlf/embedded3dviewer';
+
+    /**
+     * @access public
+     *
+     * @return void
+     */
+    public function mainAction(): void
+    {
+        if (!empty($this->requestData['model']) || !empty($this->settings['model'])) {
+            $this->view->assign('embedded3dViewerUrl', $this->buildEmbedded3dViewerUrl());
+            return;
+        }
+
+        if (!empty($this->settings['document'])) {
+            $this->assignModelFromDocument($this->getDocumentByUrl($this->settings['document']));
+        } else {
+            $this->loadDocument();
+            if (!$this->isDocMissingOrEmpty()) {
+                $this->assignModelFromDocument($this->document->getCurrentDocument());
+            }
+        }
+
+    }
+
+    /**
+     * Builds the embedded 3D viewer url.
+     *
+     * @param string $model The model url
+     * @return string The embedded 3D viewer url
+     */
+    public function buildEmbedded3dViewerUrl(string $model = ""): string
```

Comment:
```suggestion
    public function buildEmbedded3dViewerUrl(string $model = ''): string
```
It's no big deal, but please only use `"..."` for strings if this is needed for variable expansion. Using `'...'` is a tiny bit faster and avoids accidental interpretation of strings as variables.